### ISSUE TITLE
[ADE] Has anyone ever gotten this to work?

### ISSUE THREAD

**Issue by ProVega** [Azure/azure-linux-extensions#1218]
I don't know how to write this bug, but I have tried to get this feature to work on at least 10 different Ubuntu 18.04 VMs. Clean, machines with just OS disks, machines with OS + Data disks. I've bricked several machines. This thing just never works.

Latest attempt is version 1.1.0.52 - The OS disk is 32GB SSD, so I created a machine with 64GB memory. 

**Latest issue is a failure to unmount \oldroot after 10 attempts.**

I really feel like the documentation is incomplete and must be missing some critical step that I am not following. Is the issue with how I am mounting drives? With the processes that are running? 

Is there some "prep the machine" checklist that I can't find?

**Comment by ProVega** [comment]
2020/10/07 05:49:04 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: 3213: [Info] WARNING: unsupported mount scheme [/sys/kernel/debug /sys/kernel/debug/tracing]
2020/10/07 05:49:04 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: 3213: [Info] WARNING: unsupported mount scheme [/mnt /mnt/azure_bek_disk]

However, the azure_bek_disk seems to mount.

**Is this bad?  all of these packages appear to be installed.**

2020/10/07 05:50:10 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: 3712: [Info] Command apt-get install -y at cryptsetup-bin lsscsi python-parted python-six procps psmisc failed with return code 100
2020/10/07 05:50:10 stdout:
2020/10/07 05:50:10 
2020/10/07 05:50:10 stderr:
2020/10/07 05:50:10 E: Could not get lock /var/lib/dpkg/lock-frontend - open (11: Resource temporarily unavailable)
2020/10/07 05:50:10 E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?


Then later


2020/10/07 05:50:10 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: 3712: [Info] Command mountpoint /oldroot failed with return code 1
2020/10/07 05:50:10 stdout:
2020/10/07 05:50:10 
2020/10/07 05:50:10 stderr:
2020/10/07 05:50:10 mountpoint: /oldroot: No such file or directory

Is this the problem? Process in use?


2020/10/07 05:50:27 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: 3712: [Info] Command /bin/umount /usr/share/apachetomcat failed with return code 32
2020/10/07 05:50:27 stdout:
2020/10/07 05:50:27 
2020/10/07 05:50:27 stderr:
2020/10/07 05:50:27 umount: /usr/share/apachetomcat: target is busy.
2020/10/07 05:50:27 
2020/10/07 05:50:27 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: 3712: [Info] error occured when do the umount for: /usr/share/apachetomcat with code: 32
2020/10/07 05:50:27 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: Acquired Lock: /var/lib/azure_disk_encryption_config/status_lock_file.lck



Then later I get this, even though "six" is installed!


2020/10/07 05:50:28 [Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.0]: 3712: [Error] Failed to enable the extension with error: No module named six, stack trace: Traceback (most recent call last):
2020/10/07 05:50:28   File '/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.52/main/handle.py', line 2196, in daemon
2020/10/07 05:50:28     daemon_encrypt()
2020/10/07 05:50:28   File '/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.52/main/handle.py', line 1927, in daemon_encrypt
2020/10/07 05:50:28     from oscrypto.ubuntu_1604 import Ubuntu1604EncryptionStateMachine
2020/10/07 05:50:28   File '/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.52/main/oscrypto/__init__.py', line 17, in <module>
2020/10/07 05:50:28     from OSEncryptionStateMachine import *
2020/10/07 05:50:28   File '/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.52/main/oscrypto/OSEncryptionStateMachine.py', line 48, in <module>
2020/10/07 05:50:28     from transitions import *
2020/10/07 05:50:28   File '/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.52/transitions/transitions/__init__.py', line 2, in <module>
2020/10/07 05:50:28     from .core import (State, Transition, Event, EventData, Machine, MachineError,
2020/10/07 05:50:28   File '/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.52/transitions/transitions/core.py', line 8, in <module>
2020/10/07 05:50:28     from six import string_types
2020/10/07 05:50:28 ImportError: No module named six

**Comment by adyavanapalli** [comment]
Hey @ProVega, did you ever get this to work?

**Comment by ProVega** [comment]
Not really, after bricking several machines I came up with these steps that seemed to work more often then not:

Basically make sure as many services are disabled as possible and update the everything.

apt update
apt upgrade
apt autoremove

systemctl stop auoms.service
systemctl disable auoms.service
systemctl stop elasticsearch.service
systemctl disable elasticsearch.service
systemctl stop stackify-agent
systemctl disable stackify-agent
systemctl stop redis-server.service
systemctl disable redis-server.service

Required by Azure Disk Encryption
apt install grub-pc-bin
 

1. Make sure the data drive is created, attached, formatted, and mounted first. (See above)
2. Make sure you STOP and deallocate and then reallocate the VM at least once!
3. Make sure that Azure Security Center Auto Provision is disabled.
4. Logout of the VM from any SSH sessions
5. Enable encryption via CLI, this is better because the UX forces the more complex KEK wrapper key:
6. 

az vm encryption enable -g "YOUR-GROUP" --name "YOUR-SERVER" --disk-encryption-keyvault "/subscriptions/XXXXX-XXX-XXXX-XXX-XXXXXXX974/resourceGroups/YOUR-NAME-KeyVault/providers/Microsoft.KeyVault/vaults/YOUR-VAULT" --volume-type ALL --encrypt-format-all
